### PR TITLE
postgresqlPackages.postgis: enable tests

### PR DESCRIPTION
**Pull request by kirillrdy**
## Description of changes

until https://github.com/NixOS/nixpkgs/pull/331166 reaches master postgis is broken

running postgis tests does catch that regression

I am going to remove all postgresql overrides before merging, it's there for review+testing

how do people feel about nixfmt-rfc-style ?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kirillrdy** [comment]
Result of `nixpkgs-review pr 331876` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>postgresql12JitPackages.postgis</li>
    <li>postgresql12JitPackages.postgis.doc</li>
    <li>postgresql12Packages.postgis</li>
    <li>postgresql12Packages.postgis.doc</li>
    <li>postgresql13JitPackages.postgis</li>
    <li>postgresql13JitPackages.postgis.doc</li>
    <li>postgresql13Packages.postgis</li>
    <li>postgresql13Packages.postgis.doc</li>
    <li>postgresql14JitPackages.postgis</li>
    <li>postgresql14JitPackages.postgis.doc</li>
    <li>postgresql14Packages.postgis</li>
    <li>postgresql14Packages.postgis.doc</li>
    <li>postgresqlJitPackages.postgis (postgresql15JitPackages.postgis)</li>
    <li>postgresqlJitPackages.postgis.doc (postgresql15JitPackages.postgis.doc)</li>
    <li>postgresqlPackages.postgis (postgresql15Packages.postgis)</li>
    <li>postgresqlPackages.postgis.doc (postgresql15Packages.postgis.doc)</li>
    <li>postgresql16JitPackages.postgis</li>
    <li>postgresql16JitPackages.postgis.doc</li>
    <li>postgresql16Packages.postgis</li>
    <li>postgresql16Packages.postgis.doc</li>
  </ul>
</details>

**Comment by imincik** [comment]
Looks good to me and I very much support adding the tests. Good job.

**Comment by imincik** [comment]
> how do people feel about nixfmt-rfc-style ?

+1 from me.

**Review by wolfgangwalther — CHANGES_REQUESTED** [review]

File: pkgs/servers/sql/postgresql/ext/postgis.nix [hunk 1]
```diff
@@ -58,7 +62,23 @@ stdenv.mkDerivation rec {
...
+    substituteInPlace regress/run_test.pl --replace-fail "/share/contrib/postgis" "$out/share/postgresql/contrib/postgis"
+    substituteInPlace regress/Makefile --replace-fail 's,\$$libdir,$(REGRESS_INSTALLDIR)/lib,g' "s,\\$\$libdir,$PWD/regress/00-regress-install$out/lib,g" \
+      --replace-fail '$(REGRESS_INSTALLDIR)/share/contrib/postgis/*.sql' "$PWD/regress/00-regress-install$out/share/postgresql/contrib/postgis/*.sql"
+    substituteInPlace doc/postgis-out.xml --replace-fail "http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd" "${docbook_xml_dtd_45}/xml/dtd/docbook/docbookx.dtd"
+    initdb db_dir
+    pg_ctl -D db_dir -o "-k /tmp" start
+  '';
+
+  postCheck = ''
+    pg_ctl -D db_dir stop
```
[the remark this answers is not in the document]
Comment (kirillrdy) [reply pkgs/servers/sql/postgresql/ext/postgis.nix:81]: good question, lets unpin postgres to reproduce the failure, you can do it, but i will do it as well for educational purposes

[the remark this answers is not in the document]
Comment (kirillrdy) [reply pkgs/servers/sql/postgresql/ext/postgis.nix:81]: unpinning postgres produces expected failure

```
2024-08-03 08:57:11.054 UTC [9345] ERROR:  could not load library "/build/postgis-3.4.2/regress/00-regress-install/nix/store/s8rilmq4irbivy6f5g6yvd21fc9bjw8c-postgis-3.4.2/lib/postgis_raster-3.so": /nix/store/gsa6jmsac2gqyfazlfncbj251q936wqi-libspatialite-5.1.0/lib/libspatialite.so.8: undefined symbol: xmlNanoHTTPCleanup, version LIBXML2_2.4.30
```
no postres is running afterwards, I assume all sandbox processes ( cgroup??? ) gets cleaned 
```
ps aux | grep post
kirillvr   21385  0.0  0.0   6736  1536 pts/0    S+   18:58   0:00 grep post
```

[the remark this answers is not in the document]
Comment (kirillrdy) [reply pkgs/servers/sql/postgresql/ext/postgis.nix:81]: thanks for the link I was not aware of `postgresqlTestHook`

I will give it a try, as they do similar things in a more standard way

[the remark this answers is not in the document]
Comment (kirillrdy) [reply pkgs/servers/sql/postgresql/ext/postgis.nix]: changed to use postgresqlTestHook, makes it much cleaner, thank you for suggestion !

**Review by l0b0 — APPROVED** [review]

**Comment by kirillrdy** [comment]
https://github.com/NixOS/nixpkgs/pull/332764 has landed in master

**Review by imincik — APPROVED** [review]
Great thanks, looks good.

**Comment by kirillrdy** [comment]
@imincik ah sorry, i force pushed the old version :-(

**Review by imincik — APPROVED** [review]

**Pull request merged**